include/zephyr/devicetree.h

```diff
@@ -1322,6 +1323,19 @@
 #define DT_STRING_TOKEN_BY_IDX(node_id, prop, idx) \
 	DT_CAT6(node_id, _P_, prop, _IDX_, idx, _STRING_TOKEN)
 
+/**
+ * @brief Like DT_STRING_TOKEN_BY_IDX(), but with a fallback to default_value
+ * @param node_id node identifier
+ * @param prop lowercase-and-underscores property name
+ * @param idx the index to get
+ * @param default_value a fallback value to expand to
+ * @return the element in @p prop at index @p idx as a token if indexed property
+ *         is present, @p default_value otherwise
```

Comment:
Following existing conventions:
```suggestion
 * @return the element in @p prop at index @p idx as a token, or @p default_value
```